src/Emmet/Attribute.purs

```diff
@@ -0,0 +1,129 @@
+module Emmet.Attribute where
+
+import Prelude
+import DOM.HTML.Indexed.InputType (InputType, InputType(..), renderInputType) as IT
+import Data.Foldable (foldl)
+import Data.List (List, snoc)
+import Data.Maybe (Maybe(..))
+import Data.Monoid (mempty)
+import Data.Newtype (class Newtype, unwrap)
+import Data.Tuple (Tuple(..))
+
+data Attribute
+  = Class String
+  | Id String
+  | TypeInputType InputType
+  | StringAttribute String String
+  | TextContent String
+
+instance attributeEq :: Eq Attribute where
+  eq (TypeInputType a) (TypeInputType b) = eq a b
+  eq (Class a) (Class b) = eq a b
+  eq (Id a) (Id b) = eq a b
+  eq (StringAttribute a b) (StringAttribute c d) = (eq a c) && (eq b d)
+  eq _ _ = false
+
+newtype InputType = InputType IT.InputType
+derive instance newtypeInputType :: Newtype InputType _
+
+instance showInputType :: Show InputType where
+  show (InputType a) = IT.renderInputType a
+
+instance eqInputType :: Eq InputType where
```

Comment:
I'm not sure if these Eq or Show instances can be derived for the InputType wrapper because DOM.HTML.Indexed.InputType doesn't have any instances of it's own.